### PR TITLE
Feature/array functions

### PR DESCRIPTION
**Pull request by mlila**

### CODE REVIEW

File: staff/lila-petri/array-functions/concat.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * @param {Array} array First array, used as a base
+ * @param {Array} array2 Second array which will be added in the end of the first one
+ */
+
+
+function concat(array, array2) {
+    if (!(array instanceof Array)) throw TypeError(array + ' is not an array');
+    if (!(array2 instanceof Array)) throw TypeError(array2 + ' is not an array');
+   
+    var ind=array.length;
```
Comment (paulusrex) [inline staff/lila-petri/array-functions/concat.js:12]: Concat must not change the origin arrays

File: staff/lila-petri/array-functions/filter.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * @param {Function} callback The expression to evaluate.
+ */
+
+function filter(array, callback) {
+    if (!(array instanceof Array)) throw TypeError(array + ' is not an array');
+    if (typeof callback !== 'function') throw new TypeError(callback + ' is not a function');
+    var newarray=[];
+    for (var i = 0; i < array.length; i++){
+        if(callback(array[i], i, array)){
+            newarray[i]=array[i]
```
Comment (paulusrex) [inline staff/lila-petri/array-functions/filter.js]: if you do this way you will return array with holes.
function even (v) { return v % 2 === 0};
filter([4,6,9,10,5, 20,], even}
// [4,6,empty,10, empty,20]

File: staff/lila-petri/array-functions/join.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Join all the elements of the array
+ * @param {Array} array 
+ * @param {Element} separator 
+ */
+function join(array, separator){
+  if (!(array instanceof Array)) throw TypeError(array + ' is not an array');
+      var value=""
+      var i=0;
+     do{
```
Comment (paulusrex) [inline staff/lila-petri/array-functions/join.js]: ¿Y si pasas un array vacio?

File: staff/lila-petri/array-functions/concat.js (hunk 1 above)
Comment (mlila) [reply staff/lila-petri/array-functions/concat.js:12]: Done 👍

File: staff/lila-petri/array-functions/filter.js (hunk 2 above)
Comment (mlila) [reply staff/lila-petri/array-functions/filter.js]: I'm not sure if I've fixed the problem properly. Can we talk about it?

File: staff/lila-petri/array-functions/join.js (hunk 3 above)
Comment (mlila) [reply staff/lila-petri/array-functions/join.js]: Done

**Pull request merged**